src/sqlfluff/core/errors.py

```diff
@@ -83,17 +95,18 @@ def desc(self) -> str:
 
         return self.__class__.__name__  # pragma: no cover
 
-    def get_info_dict(self) -> Dict[str, Union[str, int]]:
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
```suggestion
    def to_dict(self) -> Dict[str, Union[str, int, bool]]:
```

Reply:
For the dependent classes the return value is a bit trickier, but your intent here to narrow the typing is probably sensible. I can do something which achieves that but still works for the other classes.